### PR TITLE
Sketch for parallel testing

### PR DESCRIPTION
**Pull request by krlmlr**
in preparation for parallel processing (#486).

This is still very rough, but from here on adding parallel support is almost trivial. We could use e.g. the new `future` package.

### CODE REVIEW

**Comment by krlmlr** [comment]
With multicore backend: dplyr test times drop to ~6 seconds (sequential: ~10 seconds).

File: R/test-files.R [hunk 1]
```diff
@@ -42,22 +42,36 @@ test_files <- function(paths, reporter = "summary",
     stop('No matching test file in dir')
   }
 
+  # Make sure reporter can be found before running the tests
   current_reporter <- find_reporter(reporter)
+
+  # Run without reporter first, will collect results anyway
+  if (interactive()) {
```
Comment (krlmlr) [inline R/test-files.R]: Working directory seems to be set incorrectly in parallel workers.

**Comment by hadley** [comment]
I'm not going to review for the next release since I need to get out something fairly quickly for the changes in R

**Comment by hadley** [comment]
I'm going to close this because I think the implementation will look significantly different when we use one of the higher level constructs currently under development.

**Pull request closed without merging**